### ISSUE TITLE
bug on codeblock/WITH OBJECT

### ISSUE THREAD

**Issue by JoseQuintas** [harbour/core#361]
Foto do perfil de jmcqu...@gmail.com
jmcqu...@gmail.com
26 de nov. de 2024, 15:28:18 (há 12 horas) 
para Harbour Developers
```harbour
REQUEST HB_CODEPAGE_PTISO

#include "hbclass.ch"

PROCEDURE Main()

   LOCAL a, b, c

   SetMode(25,80)
   CLS
   Set( _SET_CODEPAGE, "PTISO" )

   a := TestObj():New()
   WITH OBJECT a
      :x := 10
      c := { || :x }
   ENDWITH
   b := TestObj():New()
   WITH OBJECT b
      ? eval( c )
      ? a:x
   ENDWITH

   Inkey(0)

   RETURN

CREATE CLASS TestObj
   VAR x INIT 0
   ENDCLASS
```
Show
0
10

on first WITH OBJECT, I create codeblock with a:x
On second WITH OBJECT, codeblock returns b:x, not a:x

**Comment by dszlage** [comment]
This code does not indicate a bug in Harbour but rather illustrates the mechanism of how codeblocks work and their association with the current object.

The codeblock, defined in the first WITH OBJECT a, contains a reference to :x, which, at the time of evaluation, is bound to the current object context. When the codeblock c is executed within WITH OBJECT b, it references the :x field of that object (b), not the original object a.

The reason for this behavior is as follows:
	•	A codeblock like { || :x } is not “permanently bound” to the object in which it was defined.
	•	During the execution of EVAL(c), the codeblock refers to the object context in the current scope, which in this case is b.

If you want the codeblock to always refer to a:x, you would need to define it explicitly, for example:

c := { || a:x }

In summary, this behavior is consistent with the rules of Harbour and demonstrates the dynamic nature 
of codeblock binding to the current context. It is not a bug but a result of the language functioning as intended.
